sky/execution.py

```diff
@@ -424,6 +429,56 @@ def launch(
     entrypoint = task
     backend_utils.check_cluster_name_not_reserved(cluster_name,
                                                   operation_str='sky.launch')
+
+    if clone_disk_from:
```

Comment:
Wondering if this is best to move into _execute(), perhaps as a Stage. Pros and cons? It seems odd to add a special case treatment here.

Reply:
I don't have a strong preference for this, as this is only used by `launch`, but I moved it to the `_execution`, maybe to make it easier to read.